keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryDeliveryConfigRepository.kt

```diff
@@ -94,13 +95,22 @@ class InMemoryDeliveryConfigRepository(
 
   override fun storeConstraintState(state: ConstraintState) {
     val config = get(state.deliveryConfigName)
-    val stateId = "${state.deliveryConfigName}:${state.environmentName}:${state.artifactVersion}:" +
-      state.type
+    val idPrefix = "${state.deliveryConfigName}:${state.environmentName}:${state.artifactVersion}:"
+    val stateId = idPrefix + state.type
 
     constraints[stateId] = state
     applicationConstraintMapper
       .getOrPut(config.application, { mutableSetOf() })
       .add(stateId)
+
+    val allPass = constraints
+      .filterKeys { it.startsWith(idPrefix) }
+      .values
+      .all { it.status.passes() }
+
+    if (allPass) {
+      queueAllConstraintsApproved(state.deliveryConfigName, state.environmentName, state.artifactVersion)
+    }
```

Comment:
This covers scenarios such as submitting a manual-judgement for an older artifact version when there's already a newer version approved. It also ensures that manually overriding old failed judgement to passed results in artifact approval.